src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1872,4 +1879,128 @@ sym2.ind (λ v w, is_bridge_iff_adj_and_forall_cycle_not_mem) e
 
 end bridge_edges
 
+section induced
+
+/-! ### Connectivity of induced graphs -/
+
+lemma induce_singleton_connected (v : V) :
+  (G.induce {v}).connected :=
+begin
+  rw connected_iff,
+  refine ⟨_, by simp⟩,
+  rintro ⟨x, hx⟩ ⟨y, hy⟩,
+  rw set.mem_singleton_iff at hx hy,
+  subst_vars,
+end
+
+lemma induce_union_connected {s t : set V}
+  (sconn : (G.induce s).connected) (tconn : (G.induce t).connected) (sintert : (s ∩ t).nonempty ) :
+  (G.induce $ s ∪ t).connected :=
+begin
+  obtain ⟨u, hus, hut⟩ := sintert,
+  rw connected_iff_basepoint,
+  use ⟨u, or.inl hus⟩,
+  rintro ⟨v, hv|hv⟩,
+  { obtain ⟨p⟩ := (sconn ⟨u, hus⟩ ⟨v, hv⟩),
+    exact ⟨p.map (induce_hom_of_le le_sup_left : (G.induce s) →g (G.induce $ s ⊔ t))⟩, },
+  { obtain ⟨p⟩ := (tconn ⟨u, hut⟩ ⟨v, hv⟩),
+    exact ⟨p.map (induce_hom_of_le le_sup_right : (G.induce t) →g (G.induce $ s ⊔ t))⟩, },
+end
```

Comment:
This can also be refactored with supporting lemmas:
```lean

lemma sup_induce_le_induce_union (H : G.subgraph) (s s' : set V) :
  H.induce s ⊔ H.induce s' ≤ H.induce (s ∪ s') :=
begin
  rw sup_le_iff,
  split; mono; simp,
end

@[mono] lemma preconnected_mono {G G' : simple_graph V} (h : G ≤ G') (hc : G.preconnected) :
  G'.preconnected :=
begin
  intros v w,
  exact reachable.map (hom.map_spanning_subgraphs h) (hc v w),
end

@[mono] lemma connected_mono {G G' : simple_graph V} (h : G ≤ G') : G.connected → G'.connected :=
begin
  rw [connected_iff, connected_iff],
  exact and.imp_left (preconnected_mono h),
end

lemma subgraph.sup_connected {H H' : G.subgraph} (hH : H.connected) (hH' : H'.connected)
  (hn : (H ⊓ H').verts.nonempty) :
  (H ⊔ H').connected :=
begin
  -- let's change subgraph.connected to `@[reducible] def`.
  rw [subgraph.connected, connected_iff_basepoint],
  obtain ⟨u, hu, hu'⟩ := hn,
  use ⟨u, or.inl hu⟩,
  rintro ⟨v, hv|hv⟩,
  { exact reachable.map (subgraph.inclusion (le_sup_left : H ≤ H ⊔ H')) (hH ⟨u, hu⟩ ⟨v, hv⟩), },
  { exact reachable.map (subgraph.inclusion (le_sup_right : H' ≤ H ⊔ H')) (hH' ⟨u, hu'⟩ ⟨v, hv⟩), },
end

lemma subgraph.connected_of_connected_le (H H' : G.subgraph)
  (hle : H ≤ H') (hv : H.verts = H'.verts) (h : H.connected) : H'.connected :=
begin
  rw ← subgraph.copy_eq H' H.verts hv H'.adj rfl,
  revert h,
  apply connected_mono,
  rintro ⟨v, hv⟩ ⟨w, hw⟩ hvw,
  exact hle.2 hvw,
end

lemma subgraph.induce_union_connected {H : G.subgraph} {s t : set V}
  (sconn : (H.induce s).connected) (tconn : (H.induce t).connected) (sintert : (s ∩ t).nonempty ) :
  (H.induce $ s ∪ t).connected :=
begin
  apply subgraph.connected_of_connected_le _ _ (sup_induce_le_induce_union _ _ _),
  { simp },
  exact subgraph.sup_connected sconn tconn sintert,
end

lemma induce_union_connected {s t : set V}
  (sconn : (G.induce s).connected) (tconn : (G.induce t).connected) (sintert : (s ∩ t).nonempty ) :
  (G.induce $ s ∪ t).connected :=
begin
  rw simple_graph.induce_eq_coe_induce_top at sconn tconn ⊢,
  exact subgraph.induce_union_connected sconn tconn sintert,
end
```